### PR TITLE
arch: riscv: always set CONFIG_ARCH to 'riscv'

### PR DESCRIPTION
**Pull request by gmarull**
Note that the _dual_ `ARCH` in RISC-V is causing issues in HWMv2, hence this cleanup. Also, it didn't make much sense, since we already have `CONFIG_64BIT`.

### CODE REVIEW

**Comment by gmarull** [comment]
note: CI issue unrelated, will rebase once fixed.

**Review by henrikbrixandersen — DISMISSED** [review]

**Review by fkokosinski — DISMISSED** [review]

**Review by nordicjm — APPROVED** [review]

**Review by nashif — DISMISSED** [review]

**Comment by carlocaione** [comment]
mmmh, you sure about this? We did exactly the opposite with ARM because we had several problems keep the two together and now we are going in the opposite direction with RISCV?

**Comment by npitre** [comment]
ARM32 and ARM64 are arguably two different architectures.
Differences between RISCV32 and RISCV64 are much much smaller.

**Review by npitre — DISMISSED** [review]

**Review by soburi — DISMISSED** [review]

**Review by tejlmand — DISMISSED** [review]

**Review by dleach02 — DISMISSED** [review]

**Comment by dleach02** [comment]
and, of course, CI failure in LPC, which has nothing to do with risc plaforms, is being invoked in twister... sigh

**Comment by gmarull** [comment]
> and, of course, CI failure in LPC, which has nothing to do with risc plaforms, is being invoked in twister... sigh

We need TwisterGPT

**Review by nordicjm — CHANGES_REQUESTED** [review]
This is actually not working, don't know why CI isn't picking it up:
```
Failed processing /tmp/aa/zephyr/tests/kernel/fpu_sharing/generic   0, error:    0
Process Process-8:
Traceback (most recent call last):
  File "/usr/lib/python3.11/multiprocessing/process.py", line 314, in _bootstrap
    self.run()
  File "/usr/lib/python3.11/multiprocessing/process.py", line 108, in run
    self._target(*self._args, **self._kwargs)
  File "/tmp/aa/zephyr/scripts/pylib/twister/twisterlib/runner.py", line 1289, in pipeline_mgr
    pb.process(pipeline, done_queue, task, lock, results)
  File "/tmp/aa/zephyr/scripts/pylib/twister/twisterlib/runner.py", line 596, in process
    ret = self.cmake(filter_stages=self.instance.filter_stages)
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/tmp/aa/zephyr/scripts/pylib/twister/twisterlib/runner.py", line 1052, in cmake
    return self.run_cmake(args,filter_stages)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/tmp/aa/zephyr/scripts/pylib/twister/twisterlib/runner.py", line 401, in run_cmake
    filter_results = self.parse_generated(filter_stages)
                     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/tmp/aa/zephyr/scripts/pylib/twister/twisterlib/runner.py", line 519, in parse_generated
    raise se
  File "/tmp/aa/zephyr/scripts/pylib/twister/twisterlib/runner.py", line 514, in parse_generated
    ret = expr_parser.parse(self.testsuite.filter, filter_data, edt)
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/tmp/aa/zephyr/scripts/pylib/twister/expr_parser.py", line 313, in parse
    ast = parser.parse(expr_text)
          ^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3.11/site-packages/ply/yacc.py", line 333, in parse
    return self.parseopt_notrack(input, lexer, debug, tracking, tokenfunc)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3.11/site-packages/ply/yacc.py", line 1201, in parseopt_notrack
    tok = call_errorfunc(self.errorfunc, errtoken, self)
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3.11/site-packages/ply/yacc.py", line 192, in call_errorfunc
    r = errorfunc(token)
        ^^^^^^^^^^^^^^^^
  File "/tmp/aa/zephyr/scripts/pylib/twister/expr_parser.py", line 179, in p_error
    raise SyntaxError("Unexpected token '%s'" % p.value)
SyntaxError: Unexpected token 'CONFIG_64BIT'
```

File: tests/kernel/fpu_sharing/generic/testcase.yaml [hunk 1]
```diff
@@ -30,8 +30,8 @@ tests:
     timeout: 600
   kernel.fpu_sharing.generic.riscv32:
     extra_args: PI_NUM_ITERATIONS=500
-    filter: CONFIG_CPU_HAS_FPU
-    arch_allow: riscv32
+    filter: CONFIG_CPU_HAS_FPU && !CONFIG_64BIT
```
Comment (nordicjm) [inline tests/kernel/fpu_sharing/generic/testcase.yaml:33]:
```suggestion
    filter: CONFIG_CPU_HAS_FPU and not CONFIG_64BIT
```

File: tests/kernel/fpu_sharing/generic/testcase.yaml [hunk 2]
```diff
@@ -41,8 +41,8 @@ tests:
     extra_args: PI_NUM_ITERATIONS=500
     extra_configs:
       - CONFIG_MAIN_STACK_SIZE=2048
-    filter: CONFIG_CPU_HAS_FPU
-    arch_allow: riscv64
+    filter: CONFIG_CPU_HAS_FPU && CONFIG_64BIT
```
Comment (nordicjm) [inline tests/kernel/fpu_sharing/generic/testcase.yaml:44]:
```suggestion
    filter: CONFIG_CPU_HAS_FPU and CONFIG_64BIT
```

File: tests/kernel/fpu_sharing/generic/testcase.yaml (hunk 1 above)
Comment (gmarull) [reply tests/kernel/fpu_sharing/generic/testcase.yaml:33]: thanks, good catch

**Comment by gmarull** [comment]
last push: fixed some syntax errors, also updated patch set to be bisectable.

**Review by henrikbrixandersen — APPROVED** [review]

**Review by fkokosinski — APPROVED** [review]

**Review by nordicjm — APPROVED** [review]

**Review by fabiobaltieri — APPROVED** [review]

**Review by dleach02 — APPROVED** [review]

**Comment by fabiobaltieri** [comment]
@carlescufi we settle for 3.7 for this right?

**Review by tejlmand — APPROVED** [review]

**Pull request merged**